.changeset/modern-zebras-destroy.md

```diff
@@ -0,0 +1,28 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+`Table` - Multiple updates to the main component and its subcomponents:
+
+- Updated table headers to support tooltips
+- Updated visual treatment and location of the "sort" button in the table headers
+- Refactored CSS code to simplify usage of `hds-table`-related class names
+
+`Table::ThSort`:
+
+- Added support for tooltip via the `@tooltip` argument
+- Updated visual treatment and location of the "sort" button
+- Updated DOM structure of the `<th>` content
+- Remove class `hds-table__th-sort--button-content`
+- Replaced class `hds-table__th-sort` with classes `hds-table__th` + `hds-table__th--sort`
+- Replaced class `hds-table__th-sort--text--[left|center|right]` with `hds-table__th--align-[left|center|right]`
```

Comment:
Leaving a comment for clarification. We treated class name changes as major in [the past](https://github.com/hashicorp/design-system/blob/main/packages/components/CHANGELOG.md#200). Do we consider that past decision a one-off and moving forward we do not see class name changes as majors?

Reply:
I have mixed feelings about this. On one side yes, in the past we considered it major, but it's also true that these are not public APIs (plus, we have only one usage of these classes in the consumers codebases, in Atlas for a test selector).

@hashicorp/hds-engineering @Dhaulagiri thoughts?

----

For context, searching for the classnames usages here's what I've found:

**`hds-table__th-sort--button-content`** 
- used on the "web" repo (but not consumed, just replicating our component in React) https://github.com/search?q=%22hds-table__th-sort--button-content%22++user%3Ahashicorp+repo%3Ahashicorp%2Fweb&type=code&ref=advsearch

**`hds-table__th-sort`**
- used in Atlas for tests selectors: https://github.com/search?q=%22hds-table__th-sort%22++user%3Ahashicorp+repo%3Ahashicorp%2Fatlas&type=code&ref=advsearch
- used on the "web" repo (but not consumed, just replicating our component in React) - https://github.com/search?q=%22hds-table__th-sort%22++user%3Ahashicorp+repo%3Ahashicorp%2Fweb&type=code&ref=advsearch

**`hds-table__th--text-***`**
- used on the "web" repo (but not consumed, just replicating our component in React) - https://github.com/search?q=%22hds-table__th--text-%22++user%3Ahashicorp+repo%3Ahashicorp%2Fweb&type=code&ref=advsearch

**`hds-table__td--text-***`**
- used on the "web" repo (but not consumed, just replicating our component in React) - https://github.com/search?q=%22hds-table__td--text-%22++user%3Ahashicorp+repo%3Ahashicorp%2Fweb&type=code&ref=advsearch

Reply:
Consumers shouldn't directly be using hds component classnames (although I know there's nothing currently preventing them). Ideally we would be scoping class names though. If we've searched and confirmed that the classnames are not used in a way that would cause issues, then I wouldn't consider class name changes as major.

Reply:
Closing this conversation, since we've decided to consider this a `major` version release.